piccolo_admin/endpoints.py

```diff
@@ -482,115 +538,155 @@ def __init__(
+            "/translations/",
+            endpoint=self.get_translation_list,  # type: ignore
+            methods=["GET"],
+            tags=["Translations"],
+            response_model=TranslationListResponse,
+        )
 
-        self.auth_table = auth_table
-        self.site_name = site_name
-        self.forms = forms
-        self.read_only = read_only
-        self.sidebar_links = sidebar_links
-        self.form_config_map = {form.slug: form for form in self.forms}
+        public_app.add_api_route(
+            "/translations/{language_code:str}/",
+            endpoint=self.get_translation,  # type: ignore
+            methods=["GET"],
+            tags=["Translations"],
+            response_model=Translation,
+        )
 
-        with open(os.path.join(ASSET_PATH, "index.html")) as f:
-            self.template = f.read()
+        return public_app
 
-        #######################################################################
+    def _init_public_auth_endpoints(self, public_app: FastAPI):
+        """The unauthenticated user is supposed to hit the /login/ endpoint to obtain credentials.
+        This method modifies a FastAPI app to add /login/ and /logout/ endpoints.
+        """
+        public_app.mount(
+            path="/login/",
+            # This rate limiting is to prevent brute forcing password login,
+            # and MFA codes.
+            app=RateLimitingMiddleware(
+                app=session_login(
+                    auth_table=self._auth_table,
+                    session_table=self._session_table,
+                    session_expiry=self._session_expiry_config.session_expiry,
+                    max_session_expiry=self._session_expiry_config.max_session_expiry,
+                    redirect_to=None,
+                    production=self._production,
+                    mfa_providers=self._mfa_providers,
+                ),
+                provider=self._rate_limit_provider,
+            ),
+        )
+
+        public_app.add_route(
+            path="/logout/",
+            route=session_logout(session_table=self._session_table),  # type: ignore
+            methods=["POST"],
+        )
+
+    def _init_auth_middleware(self, api_app: FastAPI):
+        """Add middleware to a FastAPI to provide auth on all endpoints.
+        This method produces an instance of starlette's AuthenticationMiddleware and adds it to the app.
+        """
+        auth_middleware = partial(
+            AuthenticationMiddleware,
+            backend=SessionsAuthBackend(
+                auth_table=self._auth_table,
+                session_table=self._session_table,
+                admin_only=True,
+                increase_expiry=self._session_expiry_config.increase_expiry,
+            ),
+            on_error=handle_auth_exception,
+        )
+        api_app.add_middleware(auth_middleware)
 
-        private_app = FastAPI(
+    def _init_api_app(self, superuser_tables=tuple()) -> FastAPI:
```

Comment:
```suggestion
    def _init_api_app(self, superuser_tables: t.Tuple[t.Type[Table]] = tuple()) -> FastAPI:
```

I believe this type suggestion is correct, just double check